src/sensitivities/chainrules.jl

```diff
@@ -226,8 +247,12 @@ function preprocess_declaration(signature_def)
         :args => [op, :($y::Branch), ȳ, args...],
         :body => quote
             pullback = getfield($y, :pullback)  # avoid issues with getproperty overloading
-            @assert(pullback !== nothing, "pullback not set, probably because different code path used for preprocess vs for ∇. Probably need to delete a defination for ∇")
-            return pullback($ȳ)
+            if pullback === nothing
+                @debug "pullback not set, probably because different code path used for preprocess vs for ∇. Probably need to delete a defination for ∇"
+                return nothing
```

Comment:
i kinda feel like this change shouldn't have been required

Reply:
True, but this might serve as a nice way to trace future issues?

Reply:
what actions should i take here?

Reply:
Oh, nothing. I think if this package is already using the base `@debug` logging elsewhere then this should stay and we should just merge it. 👍🏻